### ISSUE TITLE
Adaptivity (coarsening) violates solute conservation

### ISSUE THREAD

**Issue by dschwen** [idaholab/moose#8349]
### Description of the enhancement or error report
If coarsening happens in a phase field simulation conserved order parameters do not have their value perfectly conserved, as the integral over the coarse element is likely to be different than the integral over the refined children.
I would like to attempt to fix this by adding source terms that exactly compensate the local non-conservation.

### Rationale for the enhancement or information for reproducing the error
Solute conservation is essential to produce reproducable results between runs with and without adaptivity. Solute drift affects the system's free energy considerably.

### Identified impact
Eliminate the unexpected solute drift. Existing simulations are unaffected.

**Comment by friedmud** [comment]
There are other other options. Instead of doing interpolation we could use
an L2 projection to get the new values. That would perfectly preserve the
concentration.

**Comment by dschwen** [comment]
Yeah, I know. Let me run this with an actual problem. I have it done and working.

**Comment by dschwen** [comment]
Wait, there is no interpolation going on when coarsening, is there? I thought nodes are just dropped. Conversely I don't see where L2 would come into play.

**Comment by YaqiWang** [comment]
With DG, projection is local thus can be easily done. Just saying.

**Comment by friedmud** [comment]
Just dropping the nodes _is_ interpolation for Lagrange...
